build.gradle

```diff
@@ -140,59 +141,59 @@ publishing {
 
         }
         Snapshot(MavenPublication) {
-        	customizePom(pom)
-        	groupId project.property('mavenGroupId')
-        	artifactId project.property('mavenArtifactId')
-        	version "${mavenMajorVersion}.${mavenMinorVersion}.${mavenPatchVersion}${mavenCentralSnapshotArtifactSuffix}"
-        	from components.java
-        	pom.withXml {
-        		def pomFile = file("${project.buildDir}/generated-pom.xml")
-        		writeTo(pomFile)
-        	}
-        	artifact sourceJar
-        	artifact javadocJar
-		}
-		
-		mavenCentralRelease(MavenPublication) {
-        	customizePom(pom)
-        	groupId project.property('mavenGroupId')
-        	artifactId project.property('mavenArtifactId')
-        	version "${mavenMajorVersion}.${mavenMinorVersion}.${mavenPatchVersion}"
-        	from components.java
-        	pom.withXml {
-        		def pomFile = file("${project.buildDir}/generated-pom.xml")
-        		writeTo(pomFile)
-        	}
-        	artifact sourceJar
-        	artifact javadocJar
-		}
+            customizePom(pom)
+            groupId project.property('mavenGroupId')
+            artifactId project.property('mavenArtifactId')
+            version "${mavenMajorVersion}.${mavenMinorVersion}.${mavenPatchVersion}${mavenCentralSnapshotArtifactSuffix}"
+            from components.java
+            pom.withXml {
+                def pomFile = file("${project.buildDir}/generated-pom.xml")
+                writeTo(pomFile)
+            }
+            artifact sourceJar
+            artifact javadocJar
+        }
+
+        mavenCentralRelease(MavenPublication) {
+            customizePom(pom)
+            groupId project.property('mavenGroupId')
+            artifactId project.property('mavenArtifactId')
+            version "${mavenMajorVersion}.${mavenMinorVersion}.${mavenPatchVersion}"
+            from components.java
+            pom.withXml {
+                def pomFile = file("${project.buildDir}/generated-pom.xml")
+                writeTo(pomFile)
+            }
+            artifact sourceJar
+            artifact javadocJar
+        }
```

Comment:
minor issue: @ramsessanchez @baywet We can consider configuring a deterministic code formatter to avoid diff due to indenting and prevent merging code that's not formatted al together.

I've worked with [spotless](https://github.com/diffplug/spotless) which can be configured to check PR's and can be used a maven / gradle plugin. For the formatting rules we can use a derivative of [google-formatter](https://github.com/google/google-java-format)

Reply:
go for it! I usually rely on editor config, but as long as it works on vs code I'm happy. Don't hesitate to pr microsoft/kiota-java (no need to PR the service libs)